MREUnityRuntime/MREUnityRuntimeLib/Core/Attachment.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MixedRealityExtension.Patching.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MixedRealityExtension.Core
+{
+    public class Attachment : IEquatable<Attachment>
+    {
+        internal string AttachPoint { get; set; }
+        internal Guid UserId { get; set; }
+
+        public bool Equals(Attachment other)
```

Comment:
For IEquatable we should also override the Object.Equals and Object.GetHashCode methods as a convention.  BehaviorHandlerBase.cs has a great example of this.

Reply:
Added a comment stating this class is not suitable for use as a hash key or dictionary key. I'm also adding an explicit override of GetHashCode in order to ward off future red flags here.

---
In reply to: [259979640](https://github.com/Microsoft/mixed-reality-extension-unity/pull/34#discussion_r259979640) [](ancestors = 259979640)

Reply:
Make sure you add the override for the Object.Equals method. So that object equality matches that of IEquatable.